include/boost/gil/channel_algorithm.hpp

```diff
@@ -352,7 +352,7 @@ template <> struct channel_convert_from_unsigned<bits16s> : public std::unary_fu
 
 template <> struct channel_convert_from_unsigned<bits32s> : public std::unary_function<bits32,bits32s> {
     typedef bits32s type;
-    type operator()(bits32 x) const { return static_cast<bits32s>(x-(1<<31)); }
+    type operator()(bits32 x) const { return static_cast<bits32s>(x)+(static_cast<bits32>(1)<<31); }
```

Comment:
This changed `-` to `+`. With unsigned 32-bit overflow semantics, the result would be the same, but the change might not have been intentional nonetheless.

Also, the way this change is applied in this particular case actually does not improve the code because you cast the previously unsigned `x` to signed and then add an unsigned value that is not representable as `bits32s`, i.e. cause `x` to be converted back to unsigned. As suggested on the ML, you should do the math on unsigned types. In this case you will have to cast to signed as the last step. This cast is technically an implementation-defined behavior if the unsigned value does not fit in the non-negative range of the target signed type, but in most cases you can hope for bitwise-equivalent result.

Reply:
@Lastique I've made the mistake last night. It should be corrected. Thank you for catching it!